app/controllers/gems_adoptions_controller.rb

```diff
@@ -0,0 +1,32 @@
+class GemsAdoptionsController < ApplicationController
+  before_action :require_login
+
+  def create
+    @gems_adoption = GemsAdoption.new(gems_adoption_params)
+    @gems_adoption.user_id = current_user.id
```

Comment:
We can do a minor but important improvement here. Build the GemsAdoption object directly from the current_user:

```ruby
@gems_adoption = current_user.gems_adoptions.build(gems_adoption_params)
```

We can do that thanks to the association between User and GemsAdoption. By building the GemsAdoption object directly from the current_user, the **user_id** attribute will be automatically set to the **id** of the current_user.